### PR TITLE
Add containerdshim interface

### PR DESCRIPTION
**Pull request by Random-Liu**
Add empty interface and priority for each function.

You can use `hack/local-up-cluster.sh` to start a containerd integrated kubernetes with all functions unimplemented.

We can implement the functions one by one based on the priority. However, the brief goal is that:
1) It's P0 to make it possible to run a pod
2) It's P1 to make whole container lifecycle work (stop, cleanup)
3) It's <=P2 to make other things work, including (streaming, logging etc.)

@mikebrow

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -184,7 +184,7 @@ KUBELET_HOST=${KUBELET_HOST:-"127.0.0.1"}
 API_CORS_ALLOWED_ORIGINS=${API_CORS_ALLOWED_ORIGINS:-/127.0.0.1(:[0-9]+)?$,/localhost(:[0-9]+)?$}
 KUBELET_PORT=${KUBELET_PORT:-10250}
 LOG_LEVEL=${LOG_LEVEL:-3}
-CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-"docker"}
+CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-"containerd"}
```
Comment (mikebrow) [inline hack/local-up-cluster.sh]: did you want to change the default here?

Comment (Random-Liu) [reply hack/local-up-cluster.sh]: Just for our development. :)

Comment (Random-Liu) [reply hack/local-up-cluster.sh]: Done.

**Comment by mikebrow** [comment]
LGTM

**Pull request merged**